### ISSUE TITLE
Offer to open a workspace from empty explorer

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#112730]
The empty explorer currently looks like this:

![image](https://user-images.githubusercontent.com/900690/102483321-d63a8a00-4064-11eb-9c80-5154207bec69.png)

I see how we want to save space and not show a "Open Workspace" button below the "Open Folder" button, but I think we could maybe leverage @sandy081 work and show a button with a dropdown?

The way this could work is:
* if you just click the button it is as today "Open Folder"
* if you click the dropdown next to it, you can also pick "Open Workspace"

If we go down this path I would think the wording should also be tweaked ("You have not yet opened a folder or workspace").

The UI component I am referring to is this:

![image](https://user-images.githubusercontent.com/900690/102483541-2fa2b900-4065-11eb-91b0-eac0b8773032.png)

**Comment by isidorn** [comment]
This makes sense. However the reason against it is to not make this any more complicated for new users who have no clue what a workspace is.
Though I am open to try this out.

@sandy081 do you have a code pointer on how to use the new UI component?

**Comment by bpasero** [comment]
Yeah agree it is a bit more advanced and requires the user to have a `*.code-workspace` file. But this is maybe a chance to get users more familar with workspaces all in all.

**Comment by sandy081** [comment]
You can use following action item to render an action with dropdown menu

https://github.com/microsoft/vscode/blob/32a18e3f5b2136db6fe6057b2a655a4489387d61/src/vs/base/browser/ui/dropdown/dropdownActionViewItem.ts#L156-L156

**Comment by isidorn** [comment]
Looking at the latest videos done by @burkeholland I am worried that this might just confuse new users additionally - what I already mentioned.

Maybe I bring this up in the UX.
If we find a nice flow for the Create Folder, maybe that is part of the drop down as well.

**Comment by misolori** [comment]
I'm still a fan of adding a split button for open folder and allowing extensions to contribute. The open folder can be: Open WSL/Codespace, Open Workspace, Clone Repo, etc. and would help reduce the amount of content that we currently have:

![image](https://user-images.githubusercontent.com/35271042/104872151-e793e200-5901-11eb-9fd3-035d500ca820.png)

We can definitely bring it up in the UX sync this week.

**Comment by isidorn** [comment]
Moving to next milestone to do it start of February to get user feedback

**Comment by misolori** [comment]
Here's a quick mockup of what this could look like:

![image](https://user-images.githubusercontent.com/35271042/105756512-97b8aa80-5f01-11eb-87db-a0d73f4a746d.png)

**Comment by isidorn** [comment]
Looks cool!

**Comment by sandy081** [comment]
In other places where I show this similar button (extensions), I also show the main action in the dropdown (from UX feedback I received). So, to be consistent, shall the dropdown also include main `Open Folder` action?

**Comment by bpasero** [comment]
Looks good to me, agree that the primary action must be the first.

**Comment by bpasero** [comment]
They should all use a "..." at the end of the label if additional user input is required to perform the action which I think is true for all.

**Comment by isidorn** [comment]
Yeah I know that pattern, I am just saying that for me personally  in a clean welcome Explorer an `Open Folder` blue button looks better than the `Open Folder...`

**Comment by isidorn** [comment]
In order to have this we would need support from the welcome views since the explorer is using the welcome view infrastructure.
The current syntax is `command:ID`, we could introduce a `commands:ID1,ID2` syntax which would mean use the dropdown action.
@joaomoreno let me know what you think?

Anyways moving to On-Deck, since we will not have time for this this milestone.

**Comment by misolori** [comment]
FYI @eamodio

**Comment by joaomoreno** [comment]
> The current syntax is `command:ID`, we could introduce a `commands:ID1,ID2` syntax which would mean use the dropdown action.

Oh man, we're growing this. Calling @jrieken for comments.

**Comment by bpasero** [comment]
I really like the idea of allowing extensions to add a button with multiple choices because that could mean the overall number of visible buttons is reduced 👍

**Comment by jrieken** [comment]
> Oh man, we're growing this. Calling @jrieken for comments.

Should be just allow to reference a submenu?

**Comment by JacksonKearl** [comment]
Current semantics are that if the link is by itself on a line, it becomes a button. We could extend this to convert a line that is exclusively links to render as a dropdown button, with the first being the main title.

```
You have not yet opened a folder

[Open Folder](command:...) [Clone Repository...](command:...) [Connect to Codespace...](command:...) [Open Workspace...](command:...) 
```

**Comment by misolori** [comment]
Bringing this back to the surface because it's getting even more crowded:

<img width="380" alt="CleanShot 2021-09-13 at 15 32 18@2x" src="https://user-images.githubusercontent.com/35271042/133164968-1b81955a-25ec-4c89-9849-7557a50a9d23.png">
